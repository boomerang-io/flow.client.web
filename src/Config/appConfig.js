// Look for the data injected into the HTML file from the Express app
// See server/app.js for implementation
export const APP_ROOT =
  window._SERVER_DATA && window._SERVER_DATA.APP_ROOT ? window._SERVER_DATA.APP_ROOT : "/BMRG_APP_ROOT_CONTEXT";

export const PRODUCT_STANDALONE =
  window._SERVER_DATA && window._SERVER_DATA.PRODUCT_STANDALONE
    ? window._SERVER_DATA.PRODUCT_STANDALONE === "true"
    : process.env.REACT_APP_PRODUCT_STANDALONE
    ? true
    : false;

export const EMBEDDED_MODE =
  window._SERVER_DATA && window._SERVER_DATA.EMBEDDED_MODE
    ? window._SERVER_DATA.EMBEDDED_MODE === "true"
    : process.env.REACT_APP_EMBEDDED_MODE
    ? true
    : true;

export const CORE_ENV_URL =
  window._SERVER_DATA && window._SERVER_DATA.CORE_ENV_URL ? window._SERVER_DATA.CORE_ENV_URL : "";

export const BASE_DOCUMENTATION_URL = PRODUCT_STANDALONE
  ? "https://www.useboomerang.io/docs/boomerang-flow"
  : `${CORE_ENV_URL}/docs/boomerang-flow`;

export const isDevEnv = process.env.NODE_ENV === "development";
export const isTestEnv = process.env.NODE_ENV === "test";

export const AppPath = {
  Root: "/",
  Error: "/error",
  Activity: "/activity",
  Editor: "/editor/:workflowId",
  EditorDesigner: `/editor/:workflowId/workflow`,
  EditorConfigure: `editor/:workflowId/configure`,
  EditorChangelog: `/editor/:workflowId/changelog`,
  EditorProperties: `/editor/:workflowId/parameters`,
  Execution: "/activity/:workflowId/execution/:executionId",
  Insights: "/insights",
  Properties: "/admin/parameters",
  Quotas: "/admin/quotas",
  QuotasEdit: "/admin/quotas/:teamId",
  Settings: "/admin/settings",
  SystemWorkflows: "/admin/system-workflows",
  TaskTemplates: "/admin/task-templates",
  TaskTemplateEdit: `/admin/task-templates/:id/:version`,
  TaskTemplateYaml: `/admin/task-templates/:id/:version/yaml-editor`,
  Team: "/admin/teams/:teamId",
  TeamSettings: "/admin/teams/:teamId/settings",
  TeamWorkflows: "/admin/teams/:teamId/workflows",
  TeamList: "/admin/teams",
  TeamProperties: `/manage/team-parameters`,
  User: "/admin/users/:userId",
  UserList: "/admin/users",

  Workflows: "/workflows",
  WorkflowsMine: "/workflows/mine",
  WorkflowsTeams: "/workflows/teams",

  ManageTaskTemplates: `/manage/task-templates`,
  ManageTaskTemplatesTeam: `/manage/task-templates/team/:teamId`,
  ManageTaskTemplateEdit: `/manage/task-templates/team/:teamId/:taskId/:version`,
  ManageTaskTemplateYaml: `/manage/task-templates/team/:teamId/:taskId/:version/yaml-editor`,
};

export const appLink = {
  activity: () => `/activity`,
  editorDesigner: ({ workflowId }) => `/editor/${workflowId}/workflow`,
  editorConfigure: ({ workflowId }) => `/editor/${workflowId}/configure`,
  editorChangelog: ({ workflowId }) => `/editor/${workflowId}/changelog`,
  editorProperties: ({ workflowId }) => `/editor/${workflowId}/parameters`,
  execution: ({ executionId, workflowId }) => `/activity/${workflowId}/execution/${executionId}`,
  insights: () => "/insights",
  properties: () => "/admin/parameters",
  manageUsers: () => "/admin/users",
  quotas: () => "/admin/quotas",
  quotasEdit: ({ teamId }) => `/admin/quotas/${teamId}`,
  settings: () => "/admin/settings",
  systemWorkflows: () => "/admin/system-workflows",
  taskTemplates: () => "/admin/task-templates",
  taskTemplateEdit: ({ id, version }) => `/admin/task-templates/${id}/${version}`,
  taskTemplateYaml: ({ id, version }) => `/admin/task-templates/${id}/${version}/yaml-editor`,
  teamProperties: () => `/manage/team-parameters`,
  teamTaskTemplates: () => `/manage/task-templates`,
  team: ({ teamId }) => `/admin/teams/${teamId}`,
  teamWorkflows: ({ teamId }) => `/admin/teams/${teamId}/workflows`,
  teamSettings: ({ teamId }) => `/admin/teams/${teamId}/settings`,
  teamList: () => "/admin/teams",
  workflows: () => "/workflows",
  workflowsMine: () => "/workflows/mine",
  workflowsTeams: () => "/workflows/teams",
  workflowActivity: ({ workflowId }) => `/activity?page=0&size=10&workflowIds=${workflowId}`,

  manageTaskTemplates: ({ teamId }) => `/manage/task-templates/team/${teamId}`,
  manageTaskTemplateEdit: ({ teamId, taskId, version }) => `/manage/task-templates/team/${teamId}/${taskId}/${version}`,
  manageTaskTemplateYaml: ({ teamId, taskId, version }) =>
    `/manage/task-templates/team/${teamId}/${taskId}/${version}/yaml-editor`,

  //external apps
  docsWorkflowEditor: () => `${BASE_DOCUMENTATION_URL}/how-to-guide/workflow-editor`,
};

export const queryStringOptions = { arrayFormat: "comma", skipEmptyString: true };

export const FeatureFlag = {
  /**
   * new Feature Flags
   */
  TeamManagementEnabled: "TeamManagementEnabled",
  WorkflowQuotasEnabled: "WorkflowQuotasEnabled",
  SettingsEnabled: "SettingsEnabled",
  UserManagementEnabled: "UserManagementEnabled",
  GlobalParametersEnabled: "GlobalParametersEnabled",
  WorkflowTokensEnabled: "WorkflowTokensEnabled",
  TaskManagerEnabled: "TaskManagerEnabled",
  EditVerifiedTasksEnabled: "EditVerifiedTasksEnabled",
  WorkflowTriggersEnabled: "WorkflowTriggersEnabled",
  TeamParametersEnabled: "TeamParametersEnabled",

  ActivityEnabled: "ActivityEnabled",
  InsightsEnabled: "InsightsEnabled",
};
