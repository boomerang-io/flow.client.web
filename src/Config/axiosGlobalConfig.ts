// Setup global config for Axios requests so they work with ICP
import axios from "axios";
axios.defaults.withCredentials = true;
