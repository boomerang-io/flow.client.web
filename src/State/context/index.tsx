import React from "react";
import { FlowTeam, FlowUser, TaskModel, WorkflowRevision, WorkflowSummary } from "Types";

export function createContext<ContextType>() {
  const context = React.createContext<ContextType | undefined>(undefined);
  function useContext() {
    const contextValue = React.useContext(context);
    if (!contextValue) throw new Error("useContext must be inside a Provider with a value");
    return contextValue;
  }
  return [useContext, context.Provider] as const;
}

export const [useAppContext, AppContextProvider] = createContext<AppContext>();

type AppContext = {
  isTutorialActive: boolean;
  communityUrl: string;
  setIsTutorialActive: (isActive: boolean) => void;
  user: FlowUser;
  teams: FlowTeam[];
};

interface taskProvider {
  category: string;
  id: string;
  icon: any;
  name: string;
}

interface stepInterface {
  activityId: string;
  duration: number;
  flowTaskStatus: string;
  id: string;
  order: number;
  outputs: any;
  startTime: string;
  taskId: string;
  taskName: string;
  runWorkflowActivityStatus: string;
}

interface workflowExecutionInterface {
  creationDate: string;
  duration: number;
  id: string;
  properties: any; //array, may need to be changed later
  status: string;
  steps: Array<stepInterface>;
  teamName: string;
  trigger: string;
  workflowId: string;
  workflowRevisionid: string;
}

interface ExecutionContext {
  tasks: Array<taskProvider>;
  workflowExecution: workflowExecutionInterface;
  workflowRevision: object;
}
export const [useExecutionContext, ExecutionContextProvider] = createContext<ExecutionContext>();

interface EditorContext {
  revisionDispatch?: Function;
  revisionState: WorkflowRevision;
  summaryData: WorkflowSummary;
  taskTemplatesData: Array<TaskModel>;
}

export const [useEditorContext, EditorContextProvider] = createContext<EditorContext>();
