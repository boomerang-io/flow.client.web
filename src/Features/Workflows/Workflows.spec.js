import React from "react";
import { Route } from "react-router-dom";
import WorkflowsHome from "./index";
import { startApiServer } from "ApiServer";
import { queryCaches } from "react-query";
import { teams, profile } from "ApiServer/fixtures";
import { AppPath, appLink } from "Config/appConfig";
import { AppContextProvider } from "State/context";

jest.mock("@boomerang-io/carbon-addons-boomerang-react", () => ({
  ...jest.requireActual("@boomerang-io/carbon-addons-boomerang-react"),
  LoadingAnimation: "LoadingAnimation",
  notify: "notify",
  Notification: "Notification",
}));

const props = {
  teamsState: {
    isFetching: false,
    status: "success",
    error: "",
    data: [],
  },
  history: {},
  importWorkflow: {},
  importWorkflowActions: {},
  onBoard: {
    show: false,
  },
};

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
  queryCaches.forEach((queryCache) => queryCache.clear());
});

describe("WorkflowsHome --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowsHome", () => {
    const { baseElement } = rtlContextRouterRender(
      <AppContextProvider
        value={{
          isTutorialActive: false,
          communityUrl: "www.ibm.com",
          setIsTutorialActive: () => {},
          user: profile,
          teams,
        }}
      >
        <Route path={AppPath.WorkflowsTeams}>
          <WorkflowsHome {...props} />
        </Route>
      </AppContextProvider>,
      { route: appLink.workflowsTeams() }
    );
    expect(baseElement).toMatchSnapshot();
  });
});
