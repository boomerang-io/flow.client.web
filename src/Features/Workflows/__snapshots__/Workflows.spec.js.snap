// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowsHome --- Snapshot Capturing Snapshot of WorkflowsHome 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="container"
    >
      <section
        class="closeHideButtons"
      >
        <button
          class="closeButton"
        >
          <svg
            aria-label="close"
            class="closeIcon"
            fill="currentColor"
            focusable="false"
            height="32"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
          <p
            class="closeText"
          >
            Don't show again
          </p>
        </button>
        <button
          class="collapseButton"
        >
          <p
            class="collapseText"
          >
            Collapse
          </p>
          <svg
            aria-label="collapse"
            class="collapseIcon"
            fill="currentColor"
            focusable="false"
            height="32"
            preserveAspectRatio="xMidYMid meet"
            role="img"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 10L26 20 24.6 21.4 16 12.8 7.4 21.4 6 20z"
            />
          </svg>
        </button>
      </section>
      <div
        class="content"
      >
        <h1
          class="title"
        >
          Welcome to your Workflows
        </h1>
        <p
          class="subtitle"
        >
          Turn ideation into Automation
        </p>
        <div
          class="buttons"
        >
          <button
            class="button"
          >
            <p
              class="buttonText"
            >
              Learn the basics
            </p>
          </button>
          <a
            class="button"
            href="slack://channel?team=T27TLPNS1&id=CFKN6FJD6"
            tabindex="0"
          >
            <p
              class="buttonText"
            >
              Slack us feedback
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <header
        class="bx--bmrg-feature-header container"
      >
        <section
          class="bx--bmrg-feature-header__content"
        >
          <div
            class="bx--bmrg-feature-header__header"
          >
            <p
              class="bx--bmrg-feature-header-title"
            >
              These are your
            </p>
            <h1
              class="bx--bmrg-feature-header-title"
            >
              
              Workflows (4)
            </h1>
          </div>
        </section>
        <div
          class="filterContainer"
        >
          <div
            class="search"
          >
            <div
              aria-labelledby="search-team-workflows-search"
              class="bx--search bx--search--xl"
              role="search"
            >
              <svg
                aria-hidden="true"
                class="bx--search-magnifier"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                />
              </svg>
              <label
                class="bx--label"
                for="search-team-workflows"
                id="search-team-workflows-search"
              >
                Search for a workflow
              </label>
              <input
                autocomplete="off"
                class="bx--search-input"
                data-testid="workflows-team-search"
                id="search-team-workflows"
                placeholder="Search for a workflow"
                role="searchbox"
                type="text"
                value=""
              />
              <button
                aria-label="Clear search input"
                class="bx--search-close bx--search-close--hidden"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="filter"
          >
            <div
              class="bx--multi-select__wrapper bx--list-box__wrapper"
            >
              <label
                class="bx--label"
                for="b-search-filter__filter"
                id="b-search-filter__filter-label"
              >
                Filter by team
              </label>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="b-search-filter__filter-label"
                class="bx--multi-select bx--combo-box bx--multi-select--filterable bx--list-box"
                role="combobox"
              >
                <div
                  aria-haspopup="true"
                  aria-labelledby="b-search-filter__filter-label"
                  class="bx--list-box__field"
                  data-toggle="true"
                  id="b-search-filter__filter"
                  role="button"
                  tabindex="-1"
                  type="button"
                >
                  <input
                    aria-autocomplete="list"
                    aria-labelledby="b-search-filter__filter-label"
                    autocomplete="off"
                    class="bx--text-input bx--text-input--empty"
                    id="b-search-filter__filter-input"
                    placeholder="Choose a team"
                    value=""
                  />
                  
                  <div
                    class="bx--list-box__menu-icon"
                  >
                    <svg
                      aria-label="Open menu"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      name="chevron--down"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                      />
                      <title>
                        Open menu
                      </title>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        aria-label="Team Workflows"
        class="content"
        role="region"
      >
        <section
          class="sectionContainer"
        >
          <hgroup
            class="header"
          >
            <h1
              class="team"
            >
              IBM Services Engineering (3)
            </h1>
          </hgroup>
          <div
            class="workflows"
          >
            <div
              class="container"
            >
              <a
                href="/editor/5eb2c4085a92d80001a16d87/workflow"
              >
                <section
                  class="details"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      alt="flow"
                      aria-hidden="true"
                      class="icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,23H11.86a4.17,4.17,0,0,0-.43-1L22,11.43A3.86,3.86,0,0,0,24,12a4,4,0,1,0-3.86-5H11.86a4,4,0,1,0,0,2h8.28a4.17,4.17,0,0,0,.43,1L10,20.57A3.86,3.86,0,0,0,8,20a4,4,0,1,0,3.86,5H20v3h8V20H20ZM8,10a2,2,0,1,1,2-2A2,2,0,0,1,8,10ZM24,6a2,2,0,1,1-2,2A2,2,0,0,1,24,6ZM8,26a2,2,0,1,1,2-2A2,2,0,0,1,8,26Zm14-4h4v4H22Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="descriptionContainer"
                  >
                    <h1
                      class="name"
                      data-testid="workflow-card-title"
                      title="ML Train – Bot Efficiency"
                    >
                      ML Train – Bot Efficiency
                    </h1>
                    <p
                      class="description"
                      title="Train and store ML model for Bot Efficiency."
                    >
                      Train and store ML model for Bot Efficiency.
                    </p>
                  </div>
                </section>
              </a>
              <section
                class="launch"
              >
                <button
                  class="bx--btn bx--btn--field bx--btn--primary"
                  tabindex="0"
                  type="button"
                >
                  Run it
                  <svg
                    aria-hidden="true"
                    aria-label="Run Workflow"
                    class="bx--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21,16a6,6,0,1,1-6,6,6,6,0,0,1,6-6m0-2a8,8,0,1,0,8,8,8,8,0,0,0-8-8Z"
                    />
                    <path
                      d="M26,4H6A2,2,0,0,0,4,6V26a2,2,0,0,0,2,2h4V26H6V12H28V6A2,2,0,0,0,26,4ZM6,10V6H26v4Z"
                    />
                    <path
                      d="M19 19L19 25 24 22 19 19z"
                    />
                  </svg>
                </button>
              </section>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Overflow card menu"
                class="bx--overflow-menu"
                style="position: absolute; right: 0px;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-label="Overflow menu icon"
                  class="bx--overflow-menu__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="8"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="24"
                    r="2"
                  />
                  <title>
                    Overflow menu icon
                  </title>
                </svg>
              </button>
            </div>
            <div
              class="container"
            >
              <a
                href="/editor/5e877e944bbc6e0001c51e6e/workflow"
              >
                <section
                  class="details"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      alt="flow"
                      aria-hidden="true"
                      class="icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,23H11.86a4.17,4.17,0,0,0-.43-1L22,11.43A3.86,3.86,0,0,0,24,12a4,4,0,1,0-3.86-5H11.86a4,4,0,1,0,0,2h8.28a4.17,4.17,0,0,0,.43,1L10,20.57A3.86,3.86,0,0,0,8,20a4,4,0,1,0,3.86,5H20v3h8V20H20ZM8,10a2,2,0,1,1,2-2A2,2,0,0,1,8,10ZM24,6a2,2,0,1,1-2,2A2,2,0,0,1,24,6ZM8,26a2,2,0,1,1,2-2A2,2,0,0,1,8,26Zm14-4h4v4H22Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="descriptionContainer"
                  >
                    <h1
                      class="name"
                      data-testid="workflow-card-title"
                      title="Java - Deploy"
                    >
                      Java - Deploy
                    </h1>
                    <p
                      class="description"
                      title=""
                    />
                  </div>
                </section>
              </a>
              <section
                class="launch"
              >
                <button
                  class="bx--btn bx--btn--field bx--btn--primary"
                  tabindex="0"
                  type="button"
                >
                  Run it
                  <svg
                    aria-hidden="true"
                    aria-label="Run Workflow"
                    class="bx--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21,16a6,6,0,1,1-6,6,6,6,0,0,1,6-6m0-2a8,8,0,1,0,8,8,8,8,0,0,0-8-8Z"
                    />
                    <path
                      d="M26,4H6A2,2,0,0,0,4,6V26a2,2,0,0,0,2,2h4V26H6V12H28V6A2,2,0,0,0,26,4ZM6,10V6H26v4Z"
                    />
                    <path
                      d="M19 19L19 25 24 22 19 19z"
                    />
                  </svg>
                </button>
              </section>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Overflow card menu"
                class="bx--overflow-menu"
                style="position: absolute; right: 0px;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-label="Overflow menu icon"
                  class="bx--overflow-menu__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="8"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="24"
                    r="2"
                  />
                  <title>
                    Overflow menu icon
                  </title>
                </svg>
              </button>
            </div>
            <div
              class="container"
            >
              <a
                href="/editor/5e877d1f4bbc6e0001c51e12/workflow"
              >
                <section
                  class="details"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      alt="flow"
                      aria-hidden="true"
                      class="icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,23H11.86a4.17,4.17,0,0,0-.43-1L22,11.43A3.86,3.86,0,0,0,24,12a4,4,0,1,0-3.86-5H11.86a4,4,0,1,0,0,2h8.28a4.17,4.17,0,0,0,.43,1L10,20.57A3.86,3.86,0,0,0,8,20a4,4,0,1,0,3.86,5H20v3h8V20H20ZM8,10a2,2,0,1,1,2-2A2,2,0,0,1,8,10ZM24,6a2,2,0,1,1-2,2A2,2,0,0,1,24,6ZM8,26a2,2,0,1,1,2-2A2,2,0,0,1,8,26Zm14-4h4v4H22Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="descriptionContainer"
                  >
                    <h1
                      class="name"
                      data-testid="workflow-card-title"
                      title="Java - Build"
                    >
                      Java - Build
                    </h1>
                    <p
                      class="description"
                      title="Java - Build"
                    >
                      Java - Build
                    </p>
                  </div>
                </section>
              </a>
              <section
                class="launch"
              >
                <button
                  class="bx--btn bx--btn--field bx--btn--primary"
                  tabindex="0"
                  type="button"
                >
                  Run it
                  <svg
                    aria-hidden="true"
                    aria-label="Run Workflow"
                    class="bx--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21,16a6,6,0,1,1-6,6,6,6,0,0,1,6-6m0-2a8,8,0,1,0,8,8,8,8,0,0,0-8-8Z"
                    />
                    <path
                      d="M26,4H6A2,2,0,0,0,4,6V26a2,2,0,0,0,2,2h4V26H6V12H28V6A2,2,0,0,0,26,4ZM6,10V6H26v4Z"
                    />
                    <path
                      d="M19 19L19 25 24 22 19 19z"
                    />
                  </svg>
                </button>
              </section>
              <div
                class="templatesWarningIcon"
              >
                <button
                  aria-describedby="icon-tooltip-1"
                  class="bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--top bx--tooltip--align-center"
                  type="button"
                >
                  <span
                    class="bx--assistive-text"
                    id="icon-tooltip-1"
                  >
                    New version of a task available! To update, edit your workflow.
                  </span>
                  <svg
                    alt="Workflow warning button"
                    class="button"
                    height="26"
                    role="button"
                    style="will-change: auto;"
                    tabindex="0"
                    viewBox="0 0 26 26"
                    width="26"
                  >
                    <title>
                      x
                    </title>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                      transform="translate(1 1)"
                    >
                      <circle
                        cx="12"
                        cy="12"
                        fill="#F9C100"
                        r="12"
                        stroke="#0072C3"
                      />
                      <path
                        d="M18.5 18h-13c-.2 0-.3-.1-.4-.2-.1-.2-.1-.3 0-.5l6.5-12c.1-.3.4-.4.6-.2.1 0 .2.1.2.2l6.5 12c.1.2.1.3 0 .5 0 .1-.2.2-.4.2zM6.3 17h11.3L12 6.5 6.3 17zm5.2-7h1v3.5h-1V10zm.5 4.8c-.4 0-.8.3-.8.8s.3.8.8.8c.4 0 .8-.3.8-.8s-.4-.8-.8-.8z"
                        fill="#000"
                      />
                    </g>
                  </svg>
                </button>
              </div>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Overflow card menu"
                class="bx--overflow-menu"
                style="position: absolute; right: 0px;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-label="Overflow menu icon"
                  class="bx--overflow-menu__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="8"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="24"
                    r="2"
                  />
                  <title>
                    Overflow menu icon
                  </title>
                </svg>
              </button>
            </div>
            <button
              class="container"
              data-testid="workflows-create-workflow-button"
            >
              <svg
                aria-hidden="true"
                class="addIcon"
                fill="currentColor"
                focusable="false"
                height="32"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                />
              </svg>
              <p
                class="text"
              >
                Create a new workflow
              </p>
            </button>
          </div>
        </section>
        <section
          class="sectionContainer"
        >
          <hgroup
            class="header"
          >
            <h1
              class="team"
            >
              IBM Services Essentials (1)
            </h1>
          </hgroup>
          <div
            class="workflows"
          >
            <div
              class="container"
            >
              <a
                href="/editor/5eb2c4085a92d80001a16d87/workflow"
              >
                <section
                  class="details"
                >
                  <div
                    class="iconContainer"
                  >
                    <svg
                      alt="flow"
                      aria-hidden="true"
                      class="icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M20,23H11.86a4.17,4.17,0,0,0-.43-1L22,11.43A3.86,3.86,0,0,0,24,12a4,4,0,1,0-3.86-5H11.86a4,4,0,1,0,0,2h8.28a4.17,4.17,0,0,0,.43,1L10,20.57A3.86,3.86,0,0,0,8,20a4,4,0,1,0,3.86,5H20v3h8V20H20ZM8,10a2,2,0,1,1,2-2A2,2,0,0,1,8,10ZM24,6a2,2,0,1,1-2,2A2,2,0,0,1,24,6ZM8,26a2,2,0,1,1,2-2A2,2,0,0,1,8,26Zm14-4h4v4H22Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="descriptionContainer"
                  >
                    <h1
                      class="name"
                      data-testid="workflow-card-title"
                      title="Essentials Training – Bot Efficiency"
                    >
                      Essentials Training – Bot Efficiency
                    </h1>
                    <p
                      class="description"
                      title="Train and store ML model for Bot Efficiency."
                    >
                      Train and store ML model for Bot Efficiency.
                    </p>
                  </div>
                </section>
              </a>
              <section
                class="launch"
              >
                <button
                  class="bx--btn bx--btn--field bx--btn--primary"
                  tabindex="0"
                  type="button"
                >
                  Run it
                  <svg
                    aria-hidden="true"
                    aria-label="Run Workflow"
                    class="bx--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="20"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21,16a6,6,0,1,1-6,6,6,6,0,0,1,6-6m0-2a8,8,0,1,0,8,8,8,8,0,0,0-8-8Z"
                    />
                    <path
                      d="M26,4H6A2,2,0,0,0,4,6V26a2,2,0,0,0,2,2h4V26H6V12H28V6A2,2,0,0,0,26,4ZM6,10V6H26v4Z"
                    />
                    <path
                      d="M19 19L19 25 24 22 19 19z"
                    />
                  </svg>
                </button>
              </section>
              <button
                aria-expanded="false"
                aria-haspopup="true"
                aria-label="Overflow card menu"
                class="bx--overflow-menu"
                style="position: absolute; right: 0px;"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-label="Overflow menu icon"
                  class="bx--overflow-menu__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="16"
                    cy="8"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="16"
                    r="2"
                  />
                  <circle
                    cx="16"
                    cy="24"
                    r="2"
                  />
                  <title>
                    Overflow menu icon
                  </title>
                </svg>
              </button>
            </div>
            <button
              class="container"
              data-testid="workflows-create-workflow-button"
            >
              <svg
                aria-hidden="true"
                class="addIcon"
                fill="currentColor"
                focusable="false"
                height="32"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                />
              </svg>
              <p
                class="text"
              >
                Create a new workflow
              </p>
            </button>
          </div>
        </section>
        
      </div>
    </div>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
